From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Philippe Coval <philippe.coval@vates.tech>
Date: Mon, 12 Oct 2026 14:18:54 +0200
Subject: [PATCH] Add XCP-ng NTP nameserver as default servers.

Currently centos nameservers are used as default, this is not desired
since centos is not involved into XCP-ng maintenance (which is a
distro on its owm).

Note pool.ntp.org's vendor zone is not yet assigned, so meanwhile
fallback to .ntp.xcp-ng.org (with hyphen).

XCP-ng is managing ntp servers using redirections (currently
{0..3}.ntp.xcp-ng.org redirects to public pool {0..3}.pool.ntp.org)

This change is applied downstream, unsure it brings any value in the
upstream since XS defined the DEFAULT_NTP_SERVERS (first arrived first
served), this could be redefined in packaging file (or doing
introspection of /etc/os-release), if becomes too difficult to
maintain.

Related-to: XCPNG-2942
Origin: https://github.com/xcp-ng/xsconsole/pull/6
Related-to: https://github.com/xcp-ng-rpms/xsconsole/pull/23
Related-to: XCPNG-3917
Signed-off-by: Philippe Coval <philippe.coval@vates.tech>

diff --git a/XSConsoleConstants.py b/XSConsoleConstants.py
index 2b98e7bd6505..4cca19442f08 100644
--- a/XSConsoleConstants.py
+++ b/XSConsoleConstants.py
@@ -1,7 +1,7 @@
 # Default NTP domains are <int>.[centos|xenserver].pool.ntp.org
 CENTOS_NTP_POOL_DOMAIN = "centos.pool.ntp.org"
-DEFAULT_NTP_DOMAINS = [CENTOS_NTP_POOL_DOMAIN, "xenserver.pool.ntp.org"]
-
+XCPNG_NTP_POOL_DOMAIN = "ntp.xcp-ng.org"
+DEFAULT_NTP_DOMAINS = [CENTOS_NTP_POOL_DOMAIN, "xenserver.pool.ntp.org", XCPNG_NTP_POOL_DOMAIN]
 NUM_DEFAULT_NTP_SERVERS = 4
 
-DEFAULT_NTP_SERVERS = ["%d.%s" % (i, CENTOS_NTP_POOL_DOMAIN) for i in range(NUM_DEFAULT_NTP_SERVERS)]
+DEFAULT_NTP_SERVERS = ["%d.%s" % (i, XCPNG_NTP_POOL_DOMAIN) for i in range(NUM_DEFAULT_NTP_SERVERS)]
